synapse/storage/databases/main/events.py

```diff
@@ -1980,6 +1978,15 @@ def _handle_redact_relations(
                 txn, self.store.get_thread_participated, (redacted_relates_to,)
             )
 
+        # Caches which might leak edits must be invalidated for the event being
```

Comment:
This feels duplicative to what happens in `_invalidate_caches_for_event`, but I think that doesn't run on the main process?

Reply:
`_invalidate_caches_for_event` happens on the main process? Otherwise the event caches on the main process would be wrong?

Reply:
I don't see any place that would be called on the main process, it only gets called via `CacheInvalidationWorkerStore.process_replication_rows` (either directly or via `_process_event_stream_row`).

I see some calls directly to `_invalidate_get_event_cache` though? These lines should actually be next to that, most likely...

Reply:
I made a change in f15a0310d8f0483ef2789e198ab972763b968c40 which consolidates the logic a bit, which gives apples-to-apples comparison of what to clear when redacting an event.